src/boat_simulator/boat_simulator/nodes/low_level_control/controller.py

```diff
@@ -0,0 +1,259 @@
+        the control target angle is reached.
+
+        Args:
+            `max_angle_range` (tuple): Max angle range individual to each different controller
+
+
+        Returns:
+            False if target control angle was not reached, else
+            True if target control angle has been reached (running error = 0)
+        """
+        error_tracker = False
+        if np.isclose(self.running_error, 0, 0.01):
+            return True
+        else:
+            change = copysign(self.control_speed * self.time_step, self.running_error)
+
+        if abs(self.running_error) > abs(change):
+            next_control = max(self.current_control_ang + change, self.max_angle_range[0])
+            self.running_error -= change
+        else:
+            next_control = max(
+                self.current_control_ang + self.running_error, self.max_angle_range[0]
+            )
+            self.running_error = 0
+            error_tracker = True
+
+        self.current_control_ang = min(next_control, self.max_angle_range[1])
+        return error_tracker
+
+
+class RudderController(ActuatorController):
+    """General Class for the Actuator Controller.
+
+    RudderController Extends: ActuatorController
+    """
+
+    def __init__(
+        self,
+        current_heading: Scalar,
+        desired_heading: Scalar,
+        current_control_ang: Scalar,
+        time_step: Scalar,
+        kp: Scalar,
+        cp: Scalar,
+        control_speed: Scalar,
+        max_angle_range=None,
+    ):
+        """Initializes the class attributes.
+
+        Args:
+            `current_heading` (Scalar): Current boat heading direction in degrees,
+             0 degrees (North) at the positive y axis and increasing clockwise.
+            `desired_heading` (Scalar): Target boating heading direction in degrees,
+             0 degrees (North) at the positive y axis and increasing clockwise.
+            `current_control_ang` (Scalar): Current control mechanism angle in degrees.
+            `time_step` (Scalar): Time per iteration given in seconds.
+            `kp` (Scalar): Proportional constant when calculating error.
+            `cp` (Scalar): Tuning parameter for control action.
+            `control_speed` (Scalar): Speed of controller change in degrees per second.
+            'running_error' (Scalar): Error between current and target control angle.
+
+        """
+
+        super().__init__(
+            current_control_ang,
+            time_step,
+            control_speed,
+            max_angle_range,
+        )
+
+        self.max_angle_range = RUDDER_MAX_ANGLE_RANGE
+        self.running_error = 0.0
+        self.current_heading = bound_to_180(current_heading)  # bound (-180, 180] in degrees
+        self.desired_heading = bound_to_180(desired_heading)  # bound (-180, 180] in degrees
+        self.kp = kp
+        self.cp = cp
+        self.setpoint = 0.0  # current setpoint angle in degrees
+        self.running_error = 0.0
```

Comment:
Running error is defined twice. Also, it's already defined in the super class and set to zero, so this isn't necessary unless you plan on changing it to something other than zero in this subclass.